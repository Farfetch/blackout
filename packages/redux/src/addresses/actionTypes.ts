/**
 * Action type dispatched when the fetch prediction request fails.
 */
export const FETCH_ADDRESS_PREDICTIONS_FAILURE =
  '@farfetch/blackout-redux/FETCH_ADDRESS_PREDICTIONS_FAILURE';
/**
 * Action type dispatched when the fetch prediction request starts.
 */
export const FETCH_ADDRESS_PREDICTIONS_REQUEST =
  '@farfetch/blackout-redux/FETCH_ADDRESS_PREDICTIONS_REQUEST';
/**
 * Action type dispatched when the fetch prediction request succeeds.
 */
export const FETCH_ADDRESS_PREDICTIONS_SUCCESS =
  '@farfetch/blackout-redux/FETCH_ADDRESS_PREDICTIONS_SUCCESS';

/**
 * Action type dispatched when the fetch prediction details request fails.
 */
export const FETCH_ADDRESS_PREDICTION_FAILURE =
  '@farfetch/blackout-redux/FETCH_ADDRESS_PREDICTION_FAILURE';
/**
 * Action type dispatched when the fetch prediction details request starts.
 */
export const FETCH_ADDRESS_PREDICTION_REQUEST =
  '@farfetch/blackout-redux/FETCH_ADDRESS_PREDICTION_REQUEST';
/**
 * Action type dispatched when the fetch prediction details request succeeds.
 */
export const FETCH_ADDRESS_PREDICTION_SUCCESS =
  '@farfetch/blackout-redux/FETCH_ADDRESS_PREDICTION_SUCCESS';

/**
 * Action type dispatched when resetting the address prediction.
 */
export const RESET_ADDRESS_PREDICTIONS =
  '@farfetch/blackout-redux/RESET_ADDRESS_PREDICTIONS';
