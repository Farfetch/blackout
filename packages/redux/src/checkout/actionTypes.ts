/**
 * Action type dispatched when the create checkout order request fails.
 */
export const CREATE_CHECKOUT_ORDER_FAILURE =
  '@farfetch/blackout-redux/CREATE_CHECKOUT_ORDER_FAILURE';
/**
 * Action type dispatched when the create checkout order request starts.
 */
export const CREATE_CHECKOUT_ORDER_REQUEST =
  '@farfetch/blackout-redux/CREATE_CHECKOUT_ORDER_REQUEST';
/**
 * Action type dispatched when the create checkout order request succeeds.
 */
export const CREATE_CHECKOUT_ORDER_SUCCESS =
  '@farfetch/blackout-redux/CREATE_CHECKOUT_ORDER_SUCCESS';

/**
 * Action type dispatched when the fetch checkout order charge request fails.
 */
export const FETCH_CHECKOUT_ORDER_CHARGE_FAILURE =
  '@farfetch/blackout-redux/FETCH_CHECKOUT_ORDER_CHARGE_FAILURE';
/**
 * Action type dispatched when the fetch checkout order charge request starts.
 */
export const FETCH_CHECKOUT_ORDER_CHARGE_REQUEST =
  '@farfetch/blackout-redux/FETCH_CHECKOUT_ORDER_CHARGE_REQUEST';
/**
 * Action type dispatched when the fetch checkout order charge request succeeds.
 */
export const FETCH_CHECKOUT_ORDER_CHARGE_SUCCESS =
  '@farfetch/blackout-redux/FETCH_CHECKOUT_ORDER_CHARGE_SUCCESS';

/**
 * Action type dispatched when the create checkout charge request fails.
 */
export const CREATE_CHECKOUT_ORDER_CHARGE_FAILURE =
  '@farfetch/blackout-redux/CREATE_CHECKOUT_ORDER_CHARGE_FAILURE';
/**
 * Action type dispatched when the create checkout order charge request starts.
 */
export const CREATE_CHECKOUT_ORDER_CHARGE_REQUEST =
  '@farfetch/blackout-redux/CREATE_CHECKOUT_ORDER_CHARGE_REQUEST';
/**
 * Action type dispatched when the create checkout order charge request succeeds.
 */
export const CREATE_CHECKOUT_ORDER_CHARGE_SUCCESS =
  '@farfetch/blackout-redux/CREATE_CHECKOUT_ORDER_CHARGE_SUCCESS';
/**
 * Action type dispatched when resetting the checkout order charge state.
 */
export const RESET_CHECKOUT_ORDER_CHARGE_STATE =
  '@farfetch/blackout-redux/RESET_CHECKOUT_ORDER_CHARGE_STATE';

/**
 * Action type dispatched when the fetch checkout request fails.
 */
export const FETCH_CHECKOUT_ORDER_FAILURE =
  '@farfetch/blackout-redux/FETCH_CHECKOUT_ORDER_FAILURE';
/**
 * Action type dispatched when the fetch checkout request starts.
 */
export const FETCH_CHECKOUT_ORDER_REQUEST =
  '@farfetch/blackout-redux/FETCH_CHECKOUT_ORDER_REQUEST';
/**
 * Action type dispatched when the fetch checkout request succeeds.
 */
export const FETCH_CHECKOUT_ORDER_SUCCESS =
  '@farfetch/blackout-redux/FETCH_CHECKOUT_ORDER_SUCCESS';

/**
 * Action type dispatched when resetting the checkout order.
 */
export const RESET_CHECKOUT = '@farfetch/blackout-redux/RESET_CHECKOUT';

/**
 * Action type dispatched when the fetch checkout details request fails.
 */
export const FETCH_CHECKOUT_ORDER_DETAILS_FAILURE =
  '@farfetch/blackout-redux/FETCH_CHECKOUT_ORDER_DETAILS_FAILURE';
/**
 * Action type dispatched when the fetch checkout details request starts.
 */
export const FETCH_CHECKOUT_ORDER_DETAILS_REQUEST =
  '@farfetch/blackout-redux/FETCH_CHECKOUT_ORDER_DETAILS_REQUEST';
/**
 * Action type dispatched when the fetch checkout details request succeeds.
 */
export const FETCH_CHECKOUT_ORDER_DETAILS_SUCCESS =
  '@farfetch/blackout-redux/FETCH_CHECKOUT_ORDER_DETAILS_SUCCESS';

/**
 * Action type dispatched when resetting the checkout order details state.
 */
export const RESET_CHECKOUT_ORDER_DETAILS_STATE =
  '@farfetch/blackout-redux/RESET_CHECKOUT_ORDER_DETAILS_STATE';

/**
 * Action type dispatched when the fetch collectpoints request fails.
 */
export const FETCH_COLLECT_POINTS_FAILURE =
  '@farfetch/blackout-redux/FETCH_COLLECT_POINTS_FAILURE';
/**
 * Action type dispatched when the fetch collectpoints request starts.
 */
export const FETCH_COLLECT_POINTS_REQUEST =
  '@farfetch/blackout-redux/FETCH_COLLECT_POINTS_REQUEST';
/**
 * Action type dispatched when the fetch collectpoints request succeeds.
 */
export const FETCH_COLLECT_POINTS_SUCCESS =
  '@farfetch/blackout-redux/FETCH_COLLECT_POINTS_SUCCESS';

/**
 * Action type dispatched when resetting the collect points state.
 */
export const RESET_COLLECT_POINTS_STATE =
  '@farfetch/blackout-redux/RESET_COLLECT_POINTS_STATE';

/**
 * Action type dispatched when the fetch item delivery provisioning request fails.
 */
export const FETCH_CHECKOUT_ORDER_DELIVERY_BUNDLE_PROVISIONING_FAILURE =
  '@farfetch/blackout-redux/FETCH_CHECKOUT_ORDER_DELIVERY_BUNDLE_PROVISIONING_FAILURE';
/**
 * Action type dispatched when the fetch item delivery provisioning request starts.
 */
export const FETCH_CHECKOUT_ORDER_DELIVERY_BUNDLE_PROVISIONING_REQUEST =
  '@farfetch/blackout-redux/FETCH_CHECKOUT_ORDER_DELIVERY_BUNDLE_PROVISIONING_REQUEST';
/**
 * Action type dispatched when the fetch item delivery provisioning request
 * succeeds.
 */
export const FETCH_CHECKOUT_ORDER_DELIVERY_BUNDLE_PROVISIONING_SUCCESS =
  '@farfetch/blackout-redux/FETCH_CHECKOUT_ORDER_DELIVERY_BUNDLE_PROVISIONING_SUCCESS';

/**
 * Action type dispatched when resetting the checkout order delivery bundle provisioning state.
 */
export const RESET_CHECKOUT_ORDER_DELIVERY_BUNDLE_PROVISIONING_STATE =
  '@farfetch/blackout-redux/RESET_CHECKOUT_ORDER_DELIVERY_BUNDLE_PROVISIONING_STATE';

/**
 * Action type dispatched when the fetch delivery bundle upgrades request fails.
 */
export const FETCH_CHECKOUT_ORDER_DELIVERY_BUNDLE_UPGRADES_FAILURE =
  '@farfetch/blackout-redux/FETCH_CHECKOUT_ORDER_DELIVERY_BUNDLE_UPGRADES_FAILURE';
/**
 * Action type dispatched when the fetch delivery bundle upgrades request starts.
 */
export const FETCH_CHECKOUT_ORDER_DELIVERY_BUNDLE_UPGRADES_REQUEST =
  '@farfetch/blackout-redux/FETCH_CHECKOUT_ORDER_DELIVERY_BUNDLE_UPGRADES_REQUEST';
/**
 * Action type dispatched when the fetch delivery bundle upgrades request succeeds.
 */
export const FETCH_CHECKOUT_ORDER_DELIVERY_BUNDLE_UPGRADES_SUCCESS =
  '@farfetch/blackout-redux/FETCH_CHECKOUT_ORDER_DELIVERY_BUNDLE_UPGRADES_SUCCESS';

/**
 * Action type dispatched when resetting the checkout order delivery bundle upgrades state.
 */
export const RESET_CHECKOUT_ORDER_DELIVERY_BUNDLE_UPGRADES_STATE =
  '@farfetch/blackout-redux/RESET_CHECKOUT_ORDER_DELIVERY_BUNDLE_UPGRADES_STATE';

/**
 * Action type dispatched when the fetch item delivery provisioning request fails.
 */
export const FETCH_CHECKOUT_ORDER_DELIVERY_BUNDLE_UPGRADE_PROVISIONING_FAILURE =
  '@farfetch/blackout-redux/FETCH_CHECKOUT_ORDER_DELIVERY_BUNDLE_UPGRADE_PROVISIONING_FAILURE';
/**
 * Action type dispatched when the fetch item delivery provisioning request starts.
 */
export const FETCH_CHECKOUT_ORDER_DELIVERY_BUNDLE_UPGRADE_PROVISIONING_REQUEST =
  '@farfetch/blackout-redux/FETCH_CHECKOUT_ORDER_DELIVERY_BUNDLE_UPGRADE_PROVISIONING_REQUEST';
/**
 * Action type dispatched when the fetch item delivery provisioning request succeeds.
 */
export const FETCH_CHECKOUT_ORDER_DELIVERY_BUNDLE_UPGRADE_PROVISIONING_SUCCESS =
  '@farfetch/blackout-redux/FETCH_CHECKOUT_ORDER_DELIVERY_BUNDLE_UPGRADE_PROVISIONING_SUCCESS';

/**
 * Action type dispatched when resetting the checkout order delivery bundle upgrade provisioning state.
 */
export const RESET_CHECKOUT_ORDER_DELIVERY_BUNDLE_UPGRADE_PROVISIONING_STATE =
  '@farfetch/blackout-redux/RESET_CHECKOUT_ORDER_DELIVERY_BUNDLE_UPGRADE_PROVISIONING_STATE';

/**
 * Action type dispatched when the fetch order operation request fails.
 */
export const FETCH_CHECKOUT_ORDER_OPERATION_FAILURE =
  '@farfetch/blackout-redux/FETCH_CHECKOUT_ORDER_OPERATION_FAILURE';
/**
 * Action type dispatched when the fetch order operation request starts.
 */
export const FETCH_CHECKOUT_ORDER_OPERATION_REQUEST =
  '@farfetch/blackout-redux/FETCH_CHECKOUT_ORDER_OPERATION_REQUEST';
/**
 * Action type dispatched when the fetch order operation request succeeds.
 */
export const FETCH_CHECKOUT_ORDER_OPERATION_SUCCESS =
  '@farfetch/blackout-redux/FETCH_CHECKOUT_ORDER_OPERATION_SUCCESS';

/**
 * Action type dispatched when resetting the checkout order operation state.
 */
export const RESET_CHECKOUT_ORDER_OPERATION_STATE =
  '@farfetch/blackout-redux/RESET_CHECKOUT_ORDER_OPERATION_STATE';

/**
 * Action type dispatched when the fetch order operations request fails.
 */
export const FETCH_CHECKOUT_ORDER_OPERATIONS_FAILURE =
  '@farfetch/blackout-redux/FETCH_CHECKOUT_ORDER_OPERATIONS_FAILURE';
/**
 * Action type dispatched when the fetch order operations request starts.
 */
export const FETCH_CHECKOUT_ORDER_OPERATIONS_REQUEST =
  '@farfetch/blackout-redux/FETCH_CHECKOUT_ORDER_OPERATIONS_REQUEST';
/**
 * Action type dispatched when the fetch order operations request succeeds.
 */
export const FETCH_CHECKOUT_ORDER_OPERATIONS_SUCCESS =
  '@farfetch/blackout-redux/FETCH_CHECKOUT_ORDER_OPERATIONS_SUCCESS';

/**
 * Action type dispatched when resetting the checkout order operations state.
 */
export const RESET_CHECKOUT_ORDER_OPERATIONS_STATE =
  '@farfetch/blackout-redux/RESET_CHECKOUT_ORDER_OPERATIONS_STATE';

/**
 * Action type dispatched when the remove checkout order item request fails.
 */
export const REMOVE_CHECKOUT_ORDER_ITEM_FAILURE =
  '@farfetch/blackout-redux/REMOVE_CHECKOUT_ORDER_ITEM_FAILURE';
/**
 * Action type dispatched when the remove checkout order item request starts.
 */
export const REMOVE_CHECKOUT_ORDER_ITEM_REQUEST =
  '@farfetch/blackout-redux/REMOVE_CHECKOUT_ORDER_ITEM_REQUEST';
/**
 * Action type dispatched when the remove checkout order item succeeds.
 */
export const REMOVE_CHECKOUT_ORDER_ITEM_SUCCESS =
  '@farfetch/blackout-redux/REMOVE_CHECKOUT_ORDER_ITEM_SUCCESS';

/**
 * Action type dispatched when resetting the remove checkout order item state.
 */
export const RESET_REMOVE_CHECKOUT_ORDER_ITEM_STATE =
  '@farfetch/blackout-redux/RESET_REMOVE_CHECKOUT_ORDER_ITEM_STATE';

/**
 * Action type dispatched when the set item tags request fails.
 */
export const SET_CHECKOUT_ORDER_ITEM_TAGS_FAILURE =
  '@farfetch/blackout-redux/SET_CHECKOUT_ORDER_ITEM_TAGS_FAILURE';
/**
 * Action type dispatched when the set item tags request starts.
 */
export const SET_CHECKOUT_ORDER_ITEM_TAGS_REQUEST =
  '@farfetch/blackout-redux/SET_CHECKOUT_ORDER_ITEM_TAGS_REQUEST';
/**
 * Action type dispatched when the set item tags request succeeds.
 */
export const SET_CHECKOUT_ORDER_ITEM_TAGS_SUCCESS =
  '@farfetch/blackout-redux/SET_CHECKOUT_ORDER_ITEM_TAGS_SUCCESS';

/**
 * Action type dispatched when resetting the checkout order item tags state.
 */
export const RESET_CHECKOUT_ORDER_ITEM_TAGS_STATE =
  '@farfetch/blackout-redux/RESET_CHECKOUT_ORDER_ITEM_TAGS_STATE';

/**
 * Action type dispatched when the set checkout order promocodes request fails.
 */
export const SET_CHECKOUT_ORDER_PROMOCODES_FAILURE =
  '@farfetch/blackout-redux/SET_CHECKOUT_ORDER_PROMOCODES_FAILURE';
/**
 * Action type dispatched when the set checkout order promocodes request starts.
 */
export const SET_CHECKOUT_ORDER_PROMOCODES_REQUEST =
  '@farfetch/blackout-redux/SET_CHECKOUT_ORDER_PROMOCODES_REQUEST';
/**
 * Action type dispatched when the set checkout order promocodes request succeeds.
 */
export const SET_CHECKOUT_ORDER_PROMOCODES_SUCCESS =
  '@farfetch/blackout-redux/SET_CHECKOUT_ORDER_PROMOCODES_SUCCESS';

/**
 * Action type dispatched when the remove checkout order promocodes request fails.
 */
export const REMOVE_CHECKOUT_ORDER_PROMOCODES_FAILURE =
  '@farfetch/blackout-redux/REMOVE_CHECKOUT_ORDER_PROMOCODES_FAILURE';
/**
 * Action type dispatched when the remove checkout order promocodes request starts.
 */
export const REMOVE_CHECKOUT_ORDER_PROMOCODES_REQUEST =
  '@farfetch/blackout-redux/REMOVE_CHECKOUT_ORDER_PROMOCODES_REQUEST';
/**
 * Action type dispatched when the remove checkout order promocodes request succeeds.
 */
export const REMOVE_CHECKOUT_ORDER_PROMOCODES_SUCCESS =
  '@farfetch/blackout-redux/REMOVE_CHECKOUT_ORDER_PROMOCODES_SUCCESS';

/**
 * Action type dispatched when resetting the checkout order promocodes state.
 */
export const RESET_CHECKOUT_ORDER_PROMOCODES_STATE =
  '@farfetch/blackout-redux/RESET_CHECKOUT_ORDER_PROMOCODES_STATE';

/**
 * Action type dispatched when the set tags request fails.
 */
export const SET_CHECKOUT_ORDER_TAGS_FAILURE =
  '@farfetch/blackout-redux/SET_CHECKOUT_ORDER_TAGS_FAILURE';
/**
 * Action type dispatched when the set tags request starts.
 */
export const SET_CHECKOUT_ORDER_TAGS_REQUEST =
  '@farfetch/blackout-redux/SET_CHECKOUT_ORDER_TAGS_REQUEST';
/**
 * Action type dispatched when the set tags request succeeds.
 */
export const SET_CHECKOUT_ORDER_TAGS_SUCCESS =
  '@farfetch/blackout-redux/SET_CHECKOUT_ORDER_TAGS_SUCCESS';

/**
 * Action type dispatched when resetting the checkout order tags state.
 */
export const RESET_CHECKOUT_ORDER_TAGS_STATE =
  '@farfetch/blackout-redux/RESET_CHECKOUT_ORDER_TAGS_STATE';

/**
 * Action type dispatched when the update checkout request fails.
 */
export const UPDATE_CHECKOUT_ORDER_FAILURE =
  '@farfetch/blackout-redux/UPDATE_CHECKOUT_ORDER_FAILURE';
/**
 * Action type dispatched when the update checkout request starts.
 */
export const UPDATE_CHECKOUT_ORDER_REQUEST =
  '@farfetch/blackout-redux/UPDATE_CHECKOUT_ORDER_REQUEST';
/**
 * Action type dispatched when the update checkout request succeeds.
 */
export const UPDATE_CHECKOUT_ORDER_SUCCESS =
  '@farfetch/blackout-redux/UPDATE_CHECKOUT_ORDER_SUCCESS';

/**
 * Action type dispatched when the update checkout order item request fails.
 */
export const UPDATE_CHECKOUT_ORDER_ITEM_FAILURE =
  '@farfetch/blackout-redux/UPDATE_CHECKOUT_ORDER_ITEM_FAILURE';
/**
 * Action type dispatched when the update checkout order item request starts.
 */
export const UPDATE_CHECKOUT_ORDER_ITEM_REQUEST =
  '@farfetch/blackout-redux/UPDATE_CHECKOUT_ORDER_ITEM_REQUEST';
/**
 * Action type dispatched when the update checkout order item request succeeds.
 */
export const UPDATE_CHECKOUT_ORDER_ITEM_SUCCESS =
  '@farfetch/blackout-redux/UPDATE_CHECKOUT_ORDER_ITEM_SUCCESS';

/**
 * Action type dispatched when resetting the update checkout order item state.
 */
export const RESET_UPDATE_CHECKOUT_ORDER_ITEM_STATE =
  '@farfetch/blackout-redux/RESET_UPDATE_CHECKOUT_ORDER_ITEM_STATE';

/**
 * Action type dispatched when the update delivery bundle upgrade request fails.
 */
export const UPDATE_CHECKOUT_ORDER_DELIVERY_BUNDLE_UPGRADE_FAILURE =
  '@farfetch/blackout-redux/UPDATE_CHECKOUT_ORDER_DELIVERY_BUNDLE_UPGRADE_FAILURE';
/**
 * Action type dispatched when the update delivery bundle upgrade request starts.
 */
export const UPDATE_CHECKOUT_ORDER_DELIVERY_BUNDLE_UPGRADE_REQUEST =
  '@farfetch/blackout-redux/UPDATE_CHECKOUT_ORDER_DELIVERY_BUNDLE_UPGRADE_REQUEST';
/**
 * Action type dispatched when the update delivery bundle upgrade request succeeds.
 */
export const UPDATE_CHECKOUT_ORDER_DELIVERY_BUNDLE_UPGRADE_SUCCESS =
  '@farfetch/blackout-redux/UPDATE_CHECKOUT_ORDER_DELIVERY_BUNDLE_UPGRADE_SUCCESS';

/**
 * Action type dispatched when the update delivery bundle upgrades request fails.
 */
export const UPDATE_CHECKOUT_ORDER_DELIVERY_BUNDLE_UPGRADES_FAILURE =
  '@farfetch/blackout-redux/UPDATE_CHECKOUT_ORDER_DELIVERY_BUNDLE_UPGRADES_FAILURE';
/**
 * Action type dispatched when the update delivery bundle upgrades request starts.
 */
export const UPDATE_CHECKOUT_ORDER_DELIVERY_BUNDLE_UPGRADES_REQUEST =
  '@farfetch/blackout-redux/UPDATE_CHECKOUT_ORDER_DELIVERY_BUNDLE_UPGRADES_REQUEST';
/**
 * Action type dispatched when the update delivery bundle upgrades request
 * succeeds.
 */
export const UPDATE_CHECKOUT_ORDER_DELIVERY_BUNDLE_UPGRADES_SUCCESS =
  '@farfetch/blackout-redux/UPDATE_CHECKOUT_ORDER_DELIVERY_BUNDLE_UPGRADES_SUCCESS';

/**
 * Action type dispatched when the update gift message request fails.
 */
export const UPDATE_CHECKOUT_ORDER_ITEMS_FAILURE =
  '@farfetch/blackout-redux/UPDATE_CHECKOUT_ORDER_ITEMS_FAILURE';
/**
 * Action type dispatched when the update gift message request starts.
 */
export const UPDATE_CHECKOUT_ORDER_ITEMS_REQUEST =
  '@farfetch/blackout-redux/UPDATE_CHECKOUT_ORDER_ITEMS_REQUEST';
/**
 * Action type dispatched when the update gift message request succeeds.
 */
export const UPDATE_CHECKOUT_ORDER_ITEMS_SUCCESS =
  '@farfetch/blackout-redux/UPDATE_CHECKOUT_ORDER_ITEMS_SUCCESS';

/**
 * Action type dispatched when resetting the checkout order items state.
 */
export const RESET_CHECKOUT_ORDER_ITEMS_STATE =
  '@farfetch/blackout-redux/RESET_CHECKOUT_ORDER_ITEMS_STATE';

/**
 * Action type dispatched when the fetch payment methods request fails.
 */
export const FETCH_CHECKOUT_ORDER_PAYMENT_METHODS_FAILURE =
  '@farfetch/blackout-redux/FETCH_CHECKOUT_ORDER_PAYMENT_METHODS_FAILURE';
/**
 * Action type dispatched when the fetch payment methods request starts.
 */
export const FETCH_CHECKOUT_ORDER_PAYMENT_METHODS_REQUEST =
  '@farfetch/blackout-redux/FETCH_CHECKOUT_ORDER_PAYMENT_METHODS_REQUEST';
/**
 * Action type dispatched when the fetch payment methods request succeeds.
 */
export const FETCH_CHECKOUT_ORDER_PAYMENT_METHODS_SUCCESS =
  '@farfetch/blackout-redux/FETCH_CHECKOUT_ORDER_PAYMENT_METHODS_SUCCESS';

/**
 * Action type dispatched when the create checkout order context request fails.
 */
export const CREATE_CHECKOUT_ORDER_CONTEXT_FAILURE =
  '@farfetch/blackout-redux/CREATE_CHECKOUT_ORDER_CONTEXT_FAILURE';
/**
 * Action type dispatched when the create checkout order context request starts.
 */
export const CREATE_CHECKOUT_ORDER_CONTEXT_REQUEST =
  '@farfetch/blackout-redux/CREATE_CHECKOUT_ORDER_CONTEXT_REQUEST';
/**
 * Action type dispatched when the create checkout order context request succeeds.
 */
export const CREATE_CHECKOUT_ORDER_CONTEXT_SUCCESS =
  '@farfetch/blackout-redux/CREATE_CHECKOUT_ORDER_CONTEXT_SUCCESS';

/**
 * Action type dispatched when resetting the checkout order contexts state.
 */
export const RESET_CHECKOUT_ORDER_CONTEXTS_STATE =
  '@farfetch/blackout-redux/RESET_CHECKOUT_ORDER_CONTEXTS_STATE';

/**
 * Action type dispatched when the fetch checkout order contexts request fails.
 */
export const FETCH_CHECKOUT_ORDER_CONTEXTS_FAILURE =
  '@farfetch/blackout-redux/FETCH_CHECKOUT_ORDER_CONTEXTS_FAILURE';
/**
 * Action type dispatched when the fetch checkout order contexts request starts.
 */
export const FETCH_CHECKOUT_ORDER_CONTEXTS_REQUEST =
  '@farfetch/blackout-redux/FETCH_CHECKOUT_ORDER_CONTEXTS_REQUEST';
/**
 * Action type dispatched when the fetch checkout order contexts request succeeds.
 */
export const FETCH_CHECKOUT_ORDER_CONTEXTS_SUCCESS =
  '@farfetch/blackout-redux/FETCH_CHECKOUT_ORDER_CONTEXTS_SUCCESS';

/**
 * Action type dispatched when resetting the checkout order context state.
 */
export const RESET_CHECKOUT_ORDER_CONTEXT_STATE =
  '@farfetch/blackout-redux/RESET_CHECKOUT_ORDER_CONTEXT_STATE';
/**
 * Action type dispatched when the fetch checkout order context request fails.
 */
export const FETCH_CHECKOUT_ORDER_CONTEXT_FAILURE =
  '@farfetch/blackout-redux/FETCH_CHECKOUT_ORDER_CONTEXT_FAILURE';
/**
 * Action type dispatched when the fetch checkout order context request starts.
 */
export const FETCH_CHECKOUT_ORDER_CONTEXT_REQUEST =
  '@farfetch/blackout-redux/FETCH_CHECKOUT_ORDER_CONTEXT_REQUEST';
/**
 * Action type dispatched when the fetch checkout order context request succeeds.
 */
export const FETCH_CHECKOUT_ORDER_CONTEXT_SUCCESS =
  '@farfetch/blackout-redux/FETCH_CHECKOUT_ORDER_CONTEXT_SUCCESS';

/**
 * Action type dispatched when the remove checkout order context request fails.
 */
export const REMOVE_CHECKOUT_ORDER_CONTEXT_FAILURE =
  '@farfetch/blackout-redux/REMOVE_CHECKOUT_ORDER_CONTEXT_FAILURE';
/**
 * Action type dispatched when the remove checkout order context request starts.
 */
export const REMOVE_CHECKOUT_ORDER_CONTEXT_REQUEST =
  '@farfetch/blackout-redux/REMOVE_CHECKOUT_ORDER_CONTEXT_REQUEST';
/**
 * Action type dispatched when the remove checkout order context request succeeds.
 */
export const REMOVE_CHECKOUT_ORDER_CONTEXT_SUCCESS =
  '@farfetch/blackout-redux/REMOVE_CHECKOUT_ORDER_CONTEXT_SUCCESS';
