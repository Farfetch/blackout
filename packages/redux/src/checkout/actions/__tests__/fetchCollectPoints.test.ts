import * as actionTypes from '../../actionTypes';
import {
  checkoutId,
  mockCollectPointsResponse,
} from 'tests/__fixtures__/checkout';
import { fetchCollectPoints } from '..';
import { getCollectPoints } from '@farfetch/blackout-client';
import { INITIAL_STATE } from '../../reducer';
import { mockStore } from '../../../../tests';
import find from 'lodash/find';

jest.mock('@farfetch/blackout-client', () => ({
  ...jest.requireActual('@farfetch/blackout-client'),
  getCollectPoints: jest.fn(),
}));

describe('fetchCollectPoints() action creator', () => {
  const checkoutMockStore = (state = {}) =>
    mockStore({ checkout: INITIAL_STATE }, state);
  const query = {
    orderId: checkoutId,
  };

  const expectedCollectPointsResult = {
    entities: {
      collectpoints: mockCollectPointsResponse,
    },
  };
  const expectedConfig = undefined;
  let store: ReturnType<typeof checkoutMockStore>;

  beforeEach(() => {
    jest.clearAllMocks();
    store = checkoutMockStore();
  });

  it('should create the correct actions for when the fetch collect points procedure fails', async () => {
    const expectedError = new Error('fetch collect points error');

    (getCollectPoints as jest.Mock).mockRejectedValueOnce(expectedError);
    expect.assertions(4);

    await fetchCollectPoints(query)(store.dispatch).catch(error => {
      expect(error).toBe(expectedError);
      expect(getCollectPoints).toHaveBeenCalledTimes(1);
      expect(getCollectPoints).toHaveBeenCalledWith(query, expectedConfig);
      expect(store.getActions()).toEqual(
        expect.arrayContaining([
          { type: actionTypes.FETCH_COLLECT_POINTS_REQUEST },
          {
            type: actionTypes.FETCH_COLLECT_POINTS_FAILURE,
            payload: { error: expectedError },
          },
        ]),
      );
    });
  });

  it('should create the correct actions for when the fetch collect points procedure is successful', async () => {
    (getCollectPoints as jest.Mock).mockResolvedValueOnce(
      mockCollectPointsResponse,
    );

    await fetchCollectPoints(query)(store.dispatch).then(clientResult => {
      expect(clientResult).toBe(mockCollectPointsResponse);
    });

    const actionResults = store.getActions();

    expect.assertions(5);
    expect(getCollectPoints).toHaveBeenCalledTimes(1);
    expect(getCollectPoints).toHaveBeenCalledWith(query, expectedConfig);
    expect(actionResults).toMatchObject([
      { type: actionTypes.FETCH_COLLECT_POINTS_REQUEST },
      {
        type: actionTypes.FETCH_COLLECT_POINTS_SUCCESS,
        meta: { id: checkoutId },
        payload: expectedCollectPointsResult,
      },
    ]);
    expect(
      find(actionResults, {
        type: actionTypes.FETCH_COLLECT_POINTS_SUCCESS,
      }),
    ).toMatchSnapshot('fetch collect points success payload');
  });
});
