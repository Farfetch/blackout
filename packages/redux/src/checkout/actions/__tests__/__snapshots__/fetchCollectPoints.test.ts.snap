// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`fetchCollectPoints() action creator should create the correct actions for when the fetch collect points procedure is successful: fetch collect points success payload 1`] = `
Object {
  "meta": Object {
    "id": 15338048,
  },
  "payload": Object {
    "entities": Object {
      "collectpoints": Array [
        Object {
          "businessDays": Array [
            Object {
              "hours": Array [
                Object {
                  "close": "17:00:00",
                  "open": "12:00:00",
                },
              ],
              "weekday": 0,
            },
          ],
          "clickAndCollect": Object {
            "collectPointId": 3195,
            "merchantLocationId": 10534,
          },
          "storeAddress": Object {
            "addressLine1": "19 Conduit Street",
            "addressLine2": "",
            "addressLine3": undefined,
            "city": Object {
              "countryId": 0,
              "id": 0,
              "name": "London",
              "stateId": undefined,
            },
            "country": Object {
              "alpha2Code": "GB",
              "alpha3Code": "GBR",
              "continentId": 3,
              "culture": "en-GB",
              "id": 215,
              "name": "United Kingdom",
              "nativeName": "United Kingdom",
              "region": "Europe",
              "regionId": 0,
              "subRegion": undefined,
              "subfolder": "/en-gb",
            },
            "ddd": undefined,
            "firstName": "ACME LONDON",
            "id": "00000000-0000-0000-0000-000000000000",
            "isCurrentBilling": false,
            "isCurrentShipping": false,
            "lastName": "",
            "location": Object {
              "componentTypeName": null,
              "latitude": "51.5121368",
              "longitude": "-0.14222780000000057",
              "name": null,
              "tag": null,
              "visualizationComponents": Object {},
            },
            "neighbourhood": undefined,
            "phone": "02074934667",
            "state": null,
            "userId": 0,
            "vatNumber": undefined,
            "zipCode": "W1S 2BH",
          },
        },
        Object {
          "businessDays": Array [
            Object {
              "hours": Array [
                Object {
                  "close": "18:00:00",
                  "open": "10:00:00",
                },
              ],
              "weekday": 1,
            },
          ],
          "clickAndCollect": Object {
            "collectPointId": 3196,
            "merchantLocationId": 10649,
          },
          "storeAddress": Object {
            "addressLine1": "833 Madison Avenue",
            "addressLine2": "",
            "addressLine3": undefined,
            "city": Object {
              "countryId": 0,
              "id": 0,
              "name": "New York",
              "stateId": undefined,
            },
            "country": Object {
              "alpha2Code": "US",
              "alpha3Code": "USA",
              "continentId": 5,
              "culture": "en-US",
              "id": 216,
              "name": "United States",
              "nativeName": "United States",
              "region": "The United States & Canada",
              "regionId": 0,
              "subRegion": undefined,
              "subfolder": "/en-us",
            },
            "ddd": undefined,
            "firstName": "ACME NEW YORK",
            "id": "00000000-0000-0000-0000-000000000000",
            "isCurrentBilling": false,
            "isCurrentShipping": false,
            "lastName": "",
            "location": Object {
              "componentTypeName": null,
              "latitude": "40.7700246",
              "longitude": "-73.96643059999997",
              "name": null,
              "tag": null,
              "visualizationComponents": Object {},
            },
            "neighbourhood": undefined,
            "phone": "+1 646 524 5401",
            "state": Object {
              "code": undefined,
              "countryId": 0,
              "id": 46,
              "name": "PT",
            },
            "userId": 0,
            "vatNumber": undefined,
            "zipCode": "10021",
          },
        },
      ],
    },
  },
  "type": "@farfetch/blackout-redux/FETCH_COLLECT_POINTS_SUCCESS",
}
`;
